cpp/src/lists/lists_column_factories.cu

```diff
@@ -16,10 +16,68 @@
 
 #include <cudf/column/column.hpp>
 #include <cudf/column/column_factories.hpp>
+#include <cudf/detail/gather.cuh>
 
 #include <rmm/cuda_stream_view.hpp>
+#include <rmm/exec_policy.hpp>
+
+#include <thrust/iterator/constant_iterator.h>
+#include <thrust/sequence.h>
 
 namespace cudf {
+namespace lists {
+namespace detail {
+
+std::unique_ptr<cudf::column> make_lists_column_from_scalar(list_scalar const& value,
+                                                            size_type size,
+                                                            rmm::cuda_stream_view stream,
+                                                            rmm::mr::device_memory_resource* mr)
+{
+  if (size == 0) {
+    return make_lists_column(0,
+                             make_empty_column(data_type{type_to_id<offset_type>()}),
+                             empty_like(value.view()),
+                             0,
+                             cudf::create_null_mask(0, mask_state::UNALLOCATED),
+                             stream,
+                             mr);
+  }
+  auto mr_final = size == 1 ? mr : rmm::mr::get_current_device_resource();
+
+  // Handcraft a 1-row column
+  auto offsets = make_numeric_column(
+    data_type{type_to_id<offset_type>()}, 2, mask_state::UNALLOCATED, stream, mr_final);
+  auto m_offsets = offsets->mutable_view();
+  thrust::sequence(rmm::exec_policy(stream),
+                   m_offsets.begin<size_type>(),
+                   m_offsets.end<size_type>(),
+                   0,
+                   value.view().size());
+  auto child           = std::make_unique<column>(value.view(), stream, mr_final);
```

Comment:
@mythrocks and I were discussing about a separate issue offline and this caught my attention - copy constructing row data from a LIST scalar can be very expensive because data under the hood has no bounded size guarantee. This is too much an overhead for `size > 1` case.

A better idea at this is just to create a 1-row column view using the view constructor, which involves no copy:
https://github.com/rapidsai/cudf/blob/92f067dcd80ce11c0431432f3817bebfa97816a0/cpp/include/cudf/column/column_view.hpp#L343-L349

For this reason, I'm relabeling this PR into `in progress`.